stage1/init/pod.go

```diff
@@ -160,6 +187,33 @@ func (p *Pod) WritePrepareAppTemplate() error {
 	return nil
 }
 
+func (p *Pod) writeAppReaper(appName string) error {
+	opts := []*unit.UnitOption{
+		unit.NewUnitOption("Unit", "Description", fmt.Sprintf("%s Reaper", appName)),
+		unit.NewUnitOption("Unit", "DefaultDependencies", "false"),
+		unit.NewUnitOption("Unit", "StopWhenUnneeded", "yes"),
+		unit.NewUnitOption("Unit", "Wants", "shutdown.service"),
+		unit.NewUnitOption("Unit", "After", "shutdown.service"),
+		unit.NewUnitOption("Unit", "Conflicts", "halt.target"),
+		unit.NewUnitOption("Unit", "Conflicts", "poweroff.target"),
```

Comment:
Should a conflict on "exit.target" be added as well? It will be needed when using systemd >= 227 with https://github.com/systemd/systemd/pull/1318